mapeditor/inspector/armywidget.cpp

```diff
@@ -0,0 +1,142 @@
+#include "armywidget.h"
+#include "ui_armywidget.h"
+#include "CCreatureHandler.h"
+
+
+ArmyWidget::ArmyWidget(CArmedInstance & a, QWidget *parent) :
+	QDialog(parent),
+	army(a),
+	ui(new Ui::ArmyWidget)
+{
+	ui->setupUi(this);
+	
+	uiCounts[0] = ui->count0; uiSlots[0] = ui->slot0;
```

Comment:
No way to render it with loop?

Reply:
I've spent some time to create macros but failed with that. It's possible to add widgets with code instead of creator but have doubts that it'll reduce amount of code

Reply:
simple widgets are ok to draw in designer, but something more complex is easier to create (and understand) from code